### PR TITLE
feat: manage page - Individual update buttons

### LINKED ISSUE #1532: manage page: add individual update buttons
Currently, we can't update apps individually, only 'update all' is available.

we should offer the option to update apps individually. to do this, the 'open' button will move to the [...] button

screenshots (also refer to figma):

Screenshot 2023-12-13 at 16 53 40

Screenshot 2023-12-13 at 16 54 59

### PR DESCRIPTION
**Pull request by codinesh**
Fixes #1532 by adding update button for each snap under updates section.

image

Hello @BLKKKBVSIK, @anasereijo, since this is similar to #1533, I have created this PR, please help with review.
1. When update all is clicked, individual snap update buttons will be disabled. This seems like different from Figma. Will make the change to match with figma if needed.

### CODE REVIEW

**Comment by BLKKKBVSIK** [comment]
Hi @codinesh !

Great work ! 
Finally, has some time to review your PR:
- The update icons aren't matching with what was planned in intern (I'll investigate that tomorrow, and add it to your PR)
- I don't think that any of the modification you made to the `pubspec.yaml` were necessary. Could you revert that?

**Comment by codinesh** [comment]
Hello @BLKKKBVSIK, thanks for reviewing the PR.

Reverted the changes made to pubspec.yaml, that change was not intended to push, it was in my local only. 

> The update icons aren't matching with what was planned in intern (I'll investigate that tomorrow, and add it to your PR)
Please let me know if I can do anything for icon change.

**Comment by BLKKKBVSIK** [comment]
Hey @codinesh,
For your PR to be tested and potentially merged, we need you to sign the [CLA](https://ubuntu.com/legal/contributors).
You can add @tim-hm (tim.holmes-mitra@canonical.com) as a contact, when signing the CLA.

**Comment by codinesh** [comment]
> You can add @tim-h

Hello @BLKKKBVSIK, I have signed the CLA, can you please retrigger the checks if possible.

**Comment by BLKKKBVSIK** [comment]
I just retriggered it, and I have the same result.
Will check what I can do. Otherwise, it will have to wait until January.

**Comment by codinesh** [comment]
> I just retriggered it, and I have the same result. Will check what I can do. Otherwise, it will have to wait until January.

thank you. I have signed a week back with tim.holmes-mitra@canonical.com as contact. Other PR CLA check ran successfully at that time. 

I have agreed to CLA again now, hope it will be resolved, please let me know if there is anything else I can do. Thanks.

**Comment by spydon** [comment]
@codinesh you have to do two things for the CLA to pass:
1. Set your primary github e-mail to the one that you signed the CLA with (if you want to be able to do web based git interactions, like updating the branch), check here how to do that: https://docs.github.com/en/github-ae@latest/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-email-preferences/setting-your-commit-email-address

2. Change the e-mail of previous commits (luckily just the last one) that have the wrong e-mail:
```
git commit --amend --author="Dinesh <codinesh@live.com>"
git push -f
```

**Comment by codinesh** [comment]
> @codinesh you have to do two things for the CLA to pass:
> 
> 1. Set your primary github e-mail to the one that you signed the CLA with (if you want to be able to do web based git interactions, like updating the branch), check here how to do that: https://docs.github.com/en/github-ae@latest/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-email-preferences/setting-your-commit-email-address
> 2. Change the e-mail of previous commits (luckily just the last one) that have the wrong e-mail ([7272358+codinesh@users.noreply.github.com](mailto:7272358+codinesh@users.noreply.github.com)):
> 
> ```
> git commit --amend --author="Dinesh <codinesh@live.com>"
> git push -f
> ```

Hello @spydon, thank you for the instructions. I have amended the commit with correct author email.

**Comment by codinesh** [comment]
> @codinesh you have to do two things for the CLA to pass:
> 
> 1. Set your primary github e-mail to the one that you signed the CLA with (if you want to be able to do web based git interactions, like updating the branch), check here how to do that: https://docs.github.com/en/github-ae@latest/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-email-preferences/setting-your-commit-email-address
> 2. Change the e-mail of previous commits (luckily just the last one) that have the wrong e-mail ([7272358+codinesh@users.noreply.github.com](mailto:7272358+codinesh@users.noreply.github.com)):
> 
> ```
> git commit --amend --author="Dinesh <codinesh@live.com>"
> git push -f
> ```

Hello @spydon, thank you for the instructions. I have ammended the commit with correct author email.

**Comment by spydon** [comment]
> Hello @spydon, thank you for the instructions. I have ammended the commit with correct author email.

Good stuff, now it passes!
I think we can review and merge this on Tuesday when we're back to work.

**Comment by codinesh** [comment]
Hello @spydon, @BLKKKBVSIK - Please let me know if there is any action on this from my end to merge this PR?

File: packages/app_center/lib/src/manage/manage_page.dart [hunk 1]
```diff
@@ -311,13 +312,14 @@ class _ActionButtons extends ConsumerWidget {
...
 
 class _ManageSnapTile extends ConsumerWidget {
-  const _ManageSnapTile({
-    required this.snap,
-    this.position = ManageTilePosition.middle,
-  });
+  const _ManageSnapTile(
+      {required this.snap,
+      this.position = ManageTilePosition.middle,
+      this.showUpdateButton = false});
```
Comment (spydon) [inline packages/app_center/lib/src/manage/manage_page.dart:318]: Trailing comma should be kept here

File: packages/app_center/pubspec.yaml [hunk 2]
```diff
@@ -66,4 +66,4 @@ flutter:
   uses-material-design: true
   generate: true
   assets:
-    - assets/
+    - assets/
```
Comment (spydon) [inline packages/app_center/pubspec.yaml:69]: New line at the end of the file should be kept according to the standard

**Review by spydon — APPROVED** [review]
Lgtm, just two minor comments

File: packages/app_center/lib/src/manage/manage_page.dart (hunk 1 above)
Comment (codinesh) [reply packages/app_center/lib/src/manage/manage_page.dart]: Thank you, will make these changes.

Comment (spydon) [reply packages/app_center/lib/src/manage/manage_page.dart]: @BLKKKBVSIK already made the changes, so it is now merged! :)

Comment (codinesh) [reply packages/app_center/lib/src/manage/manage_page.dart]: thank you, saw your commit later :)

**Comment by anasereijo** [comment]
Hi @codinesh thanks for your help!

> When update all is clicked, individual snap update buttons will be disabled. This seems like different from Figma. Will make the change to match with figma if needed.

You are correct. When the 'update all' button is clicked, the individual update buttons will be disabled, but also showing the loading status. See screenshot below:

Screenshot 2024-01-02 at 14 05 19

**Pull request merged**